packages/server/src/queue/queue.service.ts

```diff
@@ -1,14 +1,84 @@
 import { Injectable } from '@nestjs/common';
 import { Cron, CronExpression } from '@nestjs/schedule';
-import { Connection } from 'typeorm';
+import { Connection, LessThanOrEqual, MoreThanOrEqual } from 'typeorm';
 import { QueueModel } from './queue.entity';
 import { OpenQuestionStatus, ClosedQuestionStatus } from '@template/common';
 import { QuestionModel } from '../question/question.entity';
+import { OfficeHourModel } from '../course/office-hour.entity';
+
+interface TimeInterval {
+  startTime: Date;
+  endTime: Date;
+}
 
 @Injectable()
 export class QueueService {
   constructor(private connection: Connection) {}
 
+  public async addQueueTimes(queue: QueueModel) {
```

Comment:
Not sure about having this code in `QueueService`. I wonder if it'd make more sense in the QueueModel, since this code is sort of intrinsic to the queue data?